lib/msf/ui/console/command_dispatcher/auxiliary.rb

```diff
@@ -123,23 +123,25 @@ def cmd_run(*args)
       jobify = true
     end
 
-    rhosts_range = Rex::Socket::RangeWalker.new(mod.datastore['RHOSTS'])
-    unless rhosts_range && rhosts_range.length
-      print_error("Auxiliary failed: option RHOSTS failed to validate.")
-      return false
+    unless mod.datastore['RHOSTS'].nil?
```

Comment:
How about `if mod.datastore['RHOSTS']` ?

Reply:
Sure, tested and it worked fine.